udata/search/__init__.py

```diff
@@ -158,19 +158,50 @@ def register(adapter):
 
 
 class UdataFacetedSearch(FacetedSearch):
+    model = None
+    adapter = None
+
     def __init__(self, params):
-        self.sort = params.pop('sort', None)
+        self.extract_sort(params)
+        self.extract_pagination(params)
         q = params.pop('q', '')
+
         super(UdataFacetedSearch, self).__init__(q, params)
 
+    def extract_sort(self, params):
+        '''Extract and build sort query from parameters'''
+        sorts = params.pop('sort', [])
+        sorts = [sorts] if isinstance(sorts, basestring) else sorts
+        sorts = [(s[1:], 'desc')
+                 if s.startswith('-') else (s, 'asc')
+                 for s in sorts]
+        self.sorts = [
+            {self.adapter.sorts[s].field: d}
+            for s, d in sorts if s in self.adapter.sorts
+        ]
+
+    def extract_pagination(self, params):
+        '''Extract and build pagination from parameters'''
+        try:
+            self.page = max(int(params.pop('page', 1) or 1), 1)
```

Comment:
Not a huge fan of that line, proposal: `self.page = 'page' in params and int(params.pop('page')) or 1` without the `expect` that catches all.